### PR TITLE
Implement a "phase 2" API for sign_closing_transaction

### PR DESCRIPTION
**Pull request by devrandom**
This allows the signer to figure out which output is ours without potentially complicated parsing.

### CODE REVIEW

File: lightning/src/ln/chan_utils.rs [hunk 1]
```diff
@@ -80,6 +80,50 @@ pub fn build_commitment_secret(commitment_seed: &[u8; 32], idx: u64) -> [u8; 32]
...
+			witness: Vec::new(),
+		});
+		ins
+	};
+
+	let mut txouts: Vec<(TxOut, ())> = Vec::new();
+
+	if value_to_counterparty > 0 {
+		txouts.push((TxOut {
+			script_pubkey: counterparty_shutdown_script,
```
Comment (TheBlueMatt) [inline lightning/src/ln/chan_utils.rs:100]: (here and in channel.rs) Calling this the "shutdown" script is a misnomer - the payout script is different from the shutdown one.

**Review by TheBlueMatt — COMMENTED** [review]
Generally LGTM.

File: lightning/src/ln/chan_utils.rs (hunk 1 above)
Comment (devrandom) [reply lightning/src/ln/chan_utils.rs]: Done

**Review by TheBlueMatt — APPROVED** [review]

File: lightning/src/ln/chan_utils.rs (hunk 1 above)
Comment (TheBlueMatt) [reply lightning/src/ln/chan_utils.rs]: I must've been high, I'm not sure what this means. Either script or shutdown_script are fine...

File: lightning/src/ln/chan_utils.rs [hunk 2]
```diff
@@ -890,6 +890,129 @@ impl BuiltCommitmentTransaction {
 	}
 }
 
+/// This class tracks the per-transaction information needed to build a closing transaction and to
+/// actually build it and sign.
```
Comment (valentinewallace) [inline lightning/src/ln/chan_utils.rs:894]: "to build a closing transaction and to actually build it" is a bit redundant -- rephrase plz?

**Review by valentinewallace — APPROVED** [review]
Just nits, LGTM

File: lightning/src/ln/chan_utils.rs (hunk 2 above)
Comment (devrandom) [reply lightning/src/ln/chan_utils.rs:894]: This sentence structure is copied from the docs of `CommitmentTransaction`, so I'm changing it in both places.  I'm changing just a single word, please see if it makes more sense, or let me know if you have any other ideas.

Comment (valentinewallace) [reply lightning/src/ln/chan_utils.rs]: Looks good! thanks

File: lightning/src/ln/chan_utils.rs [hunk 3]
```diff
@@ -846,7 +890,130 @@ impl BuiltCommitmentTransaction {
...
+			built
+		}
+	}
+
+	/// Trust our pre-built transaction.
+	///
+	/// Applies a wrapper which allows access to the transaction.
+	///
+	/// This should only be used if you fully trust the builder of this object. It should not
+	///	be used by an external signer - instead use the verify function.
```
Comment (valentinewallace) [inline lightning/src/ln/chan_utils.rs:934]: I think there's still extra whitespace before the `be used` https://i.imgur.com/Q59rFLI.png

Comment (devrandom) [reply lightning/src/ln/chan_utils.rs]: Sorry, it took me a while to realize there's a stray tab there that makes it look different for different people.

**Review by TheBlueMatt — APPROVED** [review]
Only diff since Val's ack are addressing comments, will land after CI.

**Review by valentinewallace — APPROVED** [review]

**Pull request merged**